e2e-tests/tfm/cypress/fixtures/facility-tenor.js

```diff
@@ -0,0 +1,41 @@
+import { add } from 'date-fns';
+import { oneMonth, twoYearsAgo } from '../../../e2e-fixtures/dateConstants';
+
+/**
+ * Function to calculate facility tenor for a BSS facility with a commencement date
+ * two years ago and an expiry date one month from today.
+ *
+ * Various tests involve a BSS deal with a facility that has a commencement
+ * date two years ago today and an expiry date one month from today.
+ *
+ * The tenor is calculated by MDM using a SQL function.
+ *
+ * The tenor is the difference in months between the expiry date and the commencement date.
+ * But in some cases we add one.
+ *
+ * Specifically in the case of these tests, the tenor is 25 months except for:
+ * - if expiry and commencement dates are the same date of month then you add one
+ * - if commencement is end of month and expiry is also end of month you add one
+ */
+const calculateFacilityTenorForBssFacilityWithCommencementDateTwoYearsAgoAndExpiryDateOneMonthFromToday = () => {
```

Comment:
this is far too long.  Can this just be calculateBSSFacilityTenorFromCommencementDate or just calculateBSSFacilityTenor

Reply:
Problem is I then need to pass in the commencement daet, but this logic does not hold up for any commencement date - requires more ifs etc to handle any pair of dates

Reply:
And it is not a priority to write a function that accepts any pair of dates since is not needed and so would be a waste of time right now

Reply:
this is only being used for one purpose right now and there is no plan to add more.  A much shorter name is required here.  This can be updated in the future where needed (but I do not think it will ever be needed anytime soon)

Reply:
I disagree - this is duplicating logic from an entirely separate codebase. Any new devs trying to maintain this if they ever add in a new test using facility tenor they might think 'nice I can use this existing function to calculate the tenor' and then it be a real headache and confusing when it doesn't work.

If you still disagree, let me know and I will change it, I guess it is very unlikely to be me dealing with any future maintenance.

Reply:
* This function deals with both the classic deals `BSS/EWCS` so the function does not represent the true state as it stands, I would highly recommend shortering the function naming any additional function logic is already well written in the DocString.

Reply:
I don't know so much about this area so correct me if I am wrong, but the SQL that this is mirroring, which I traced through the repos and is where MDM calculates the tenor, has logic where if 'BS' is passed in we add an extra day in some cases where we don't if 'EW' is passed in. I have assumed BS -> BSS and EW -> EWCS but please do correct me if I am wrong.

Reply:
I will shorten the name however since you are both in favour